File: DSA-2026/ARRAY/Two Pointers/JS/maxProfit.js
var maxProfit = function(prices) {
    
    
    let l=0;
    let maxProfit= 0;
    for(let i =0;i<prices.length;i++){
        maxProfit = Math.max(maxProfit, prices[i]-prices[l]);
        if(prices[i]<prices[l]) l=i;

    }
    return maxProfit
};

File: DSA-2026/ARRAY/Two Pointers/JAVA/maxProfit.java


public class maxProfit {
    public int maxProfits(int[] prices) {
        int max=0;
        int l=0;
        for(int i =0;i<prices.length;i++){
            max=Math.max(max,prices[i]-prices[l]);
            if(prices[i]<prices[l]) {
                l=i;
            }
        }
        return max;
        
    }
}